src/app/pages/IdxPage/index.test.jsx

```diff
@@ -1,27 +1,36 @@
 import React from 'react';
-import { render } from '@testing-library/react';
-import IdxPage from '.';
+import { render, act } from '@testing-library/react';
+import dariRadioScheduleData from '#data/persian/bbc_dari_radio/schedule.json';
 import persianAfghanistanIdxData from '#data/persian/afghanistan';
-import { ServiceContextProvider } from '#contexts/ServiceContext';
-
-const IdxPageWithContext = () => (
-  <ServiceContextProvider service="persian">
-    <IdxPage pageData={persianAfghanistanIdxData} />
-  </ServiceContextProvider>
-);
+import IdxPageWithContext from './testHelpers';
 
 describe('IdxPage', () => {
+  beforeEach(async () => {
+    fetch.mockResponse(JSON.stringify(dariRadioScheduleData));
+  });
+
   describe('Snapshots', () => {
-    it('should render an IDX page correctly', () => {
-      const container = render(<IdxPageWithContext />);
+    it('should render a persian IDX page correctly with radio schedule', async () => {
+      let container = null;
+      await act(async () => {
+        container = render(
+          <IdxPageWithContext pageData={persianAfghanistanIdxData} />,
+        ).container;
+      });
+
       expect(container).toMatchSnapshot();
     });
 
     it('should render idx page sections', async () => {
```

Comment:
Can you add another `describe` here to separate the `snapshots` from the `assertions`?

Reply:
Done now together with other suggestions:)

Reply:
Oh yeah, I remember now 👍